src/cli/onefuzz/debug.py

```diff
@@ -434,21 +470,57 @@ def keyword(
         :param bool raw: Do not simplify the data result
         """
 
-        # See https://docs.microsoft.com/en-us/azure/data-explorer/kql-quick-reference
+        components = self._build_keyword_query(value, limit=limit)
 
-        components = ["union isfuzzy=true exceptions, traces, customEvents"]
+        return self._query_parts(components, timespan=timespan, raw=raw)
 
-        value = value.strip()
-        keywords = ['* has "%s"' % (x.replace('"', '\\"')) for x in value.split(" ")]
-        if keywords:
-            components.append("where " + " and ".join(keywords))
+    def tail(
+        self,
+        value: str,
+        *,
+        limit: int = 1000,
+        indent: Optional[int] = None,
+        filter: Optional[str] = "[message, name, customDimensions]",
+        timespan: Optional[str] = HOUR_TIMESPAN,
+    ) -> None:
+        """
+        Perform an Application Insights keyword query akin to "Transaction Search"
 
-        components.append("order by timestamp desc")
+        :param str value: Keyword to query Application Insights
+        :param str indent: Specify indent for JSON printing
+        :param str limit: Limit the number of records to return in each query
+        :param str filter: JMESPath filter for streaming results
+        """
 
-        if limit is not None:
-            components.append(f"take {limit}")
+        expression = None
+        if filter:
+            expression = jmespath.compile(filter)
 
-        return self._query_parts(components, timespan=timespan, raw=raw)
+        base_query = self._build_keyword_query(value, limit=limit, desc=False)
+
+        last_seen: Optional[str] = None
+        wait = 10.0
+
+        while True:
+            query = base_query.copy()
+            if last_seen is not None:
+                query.append(f'where timestamp > datetime("{last_seen}")')
+            results = self._query_parts(query, timespan=timespan)
+
+            if results:
+                last_seen = results[-1]["timestamp"]
+                for entry in results:
+                    if expression is not None:
+                        entry = expression.search(entry)
+                    if entry:
+                        print(json.dumps(entry, indent=indent, sort_keys=True))
+                wait = 10
```

Comment:
This is an `int`, though `wait` is meant to be a `float`. This issue will disappear if we use a constant for resetting the `wait` value.